### PR TITLE
World gen test

### PR DESCRIPTION
**Pull request by Mike-at-TRU**

### CODE REVIEW

**Review by Jaxtin-L — COMMENTED** [review]
you did so good

File: src/generated/resources/data/trutermproject/advancement/recipes/misc/beryl_from_smelting_raw_beryl.json [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+      "has_raw_beryl"
     ]
   ],
   "rewards": {
     "recipes": [
-      "trutermproject:beryl_from_smelting_beryl"
+      "trutermproject:beryl_from_smelting_raw_beryl"
     ]
   }
 }
```
Comment (aaron-plahn) [inline src/generated/resources/data/trutermproject/advancement/recipes/misc/beryl_from_smelting_raw_beryl.json:32]: Do you have an auto-formatter to add a new end of line character on save?

File: src/generated/resources/data/trutermproject/neoforge/biome_modifier/add_beryl_ore.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (aaron-plahn) [inline src/generated/resources/data/trutermproject/neoforge/biome_modifier/add_beryl_ore.json:1]: I don't have a lot to say about the `json` files as I don't fully have context for how they work.

File: src/main/java/net/tru/trutermproject/datagen/DataGenerators.java [hunk 3]
```diff
@@ -42,6 +43,7 @@ public static void gatherData(GatherDataEvent event) {
 
         generator.addProvider(event.includeClient(), new ModItemModelProvider(packOutput, existingFileHelper));
```
Comment (aaron-plahn) [inline src/main/java/net/tru/trutermproject/datagen/DataGenerators.java:44]: This seems simple enough.

File: src/main/java/net/tru/trutermproject/datagen/WorldGen/ModBiomeModifiers.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                        GenerationStep.Decoration.UNDERGROUND_ORES)
+        );
+
+    }
+
+    private static ResourceKey<BiomeModifier> createKey(String name) {
+        return ResourceKey.create(NeoForgeRegistries.Keys.BIOME_MODIFIERS,
+                ResourceLocation.fromNamespaceAndPath(TRUTermProject.MOD_ID, name));
+    }
+}
```
Comment (aaron-plahn) [inline src/main/java/net/tru/trutermproject/datagen/WorldGen/ModBiomeModifiers.java:40]: Auto format on save?

File: src/main/java/net/tru/trutermproject/datagen/WorldGen/ModBiomeModifiers.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.levelgen.GenerationStep;
+import net.minecraft.world.level.levelgen.placement.PlacedFeature;
+import net.neoforged.neoforge.common.world.BiomeModifier;
+import net.neoforged.neoforge.common.world.BiomeModifiers;
+import net.neoforged.neoforge.registries.NeoForgeRegistries;
+import net.tru.trutermproject.TRUTermProject;
+import net.tru.trutermproject.datagen.WorldGen.Ore.ModPlacedFeatures;
+
+public class ModBiomeModifiers {
```
Comment (aaron-plahn) [inline src/main/java/net/tru/trutermproject/datagen/WorldGen/ModBiomeModifiers.java:19]: Looks good.

File: src/main/java/net/tru/trutermproject/datagen/WorldGen/ModWorldGenProvider.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.concurrent.CompletableFuture;
+
+public class ModWorldGenProvider extends DatapackBuiltinEntriesProvider {
+    public ModWorldGenProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> registries) {
+        super(output, registries, new RegistrySetBuilder()
+                        .add(Registries.CONFIGURED_FEATURE, ModConfiguredFeatures::bootstrap)
+                        .add(Registries.PLACED_FEATURE, ModPlacedFeatures::bootstrap)
+                        .add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, ModBiomeModifiers::bootstrap),
+                Set.of(TRUTermProject.MOD_ID));
+
```
Comment (aaron-plahn) [inline src/main/java/net/tru/trutermproject/datagen/WorldGen/ModWorldGenProvider.java:25]: Be sure to autoformat to remove extra white space.

File: src/main/java/net/tru/trutermproject/datagen/WorldGen/Ore/ModPlacedFeatures.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
+import net.minecraft.world.level.levelgen.placement.HeightRangePlacement;
+import net.minecraft.world.level.levelgen.placement.PlacedFeature;
+import net.minecraft.world.level.levelgen.placement.PlacementModifier;
+import net.tru.trutermproject.TRUTermProject;
+import org.joml.Vector2dc;
+
+import java.util.List;
+
+//stolen from https://www.youtube.com/watch?v=_7sc7INW0IY
```
Comment (aaron-plahn) [inline src/main/java/net/tru/trutermproject/datagen/WorldGen/Ore/ModPlacedFeatures.java:19]: You might want to reword this as:
```
This was inspired by the following YouTube tutorial. <Note about customization if applicable>
```

**Review by aaron-plahn — APPROVED** [review]
Be sure to address the formatting issues before merging.
